File: PHP/6php 2/tribonacci.php
<?php
/*
|--------------------------------------------------------------------------
| Tribonacci Sequence
|--------------------------------------------------------------------------
|
*/

function tribonacci($signature, $n) {
   if ($n === 0) return [];
   $res = array_slice($signature, 0, $n);
   while (count($res) < $n) {
      $x = array_sum(array_slice($res, -3));
      array_push($res, $x);
   }
   return $res;
}


function tribonacci($signature, $n) {
   if (sizeof($signature) >= $n) {
      return array_splice($signature, 0, $n);
   }
   $signature[] = array_sum(array_slice($signature, -3)); 
   return tribonacci($signature, $n);
}


function tribonacci(array $signature, int $n): array {
   for ($i = $n - 3; $i > 0; $i--) {
      $signature[] = array_sum(array_slice($signature, -3));
   }
   return array_slice($signature, 0, $n);
}


print_r(tribonacci([1, 1, 1], 10));
print_r(tribonacci([0, 0, 1], 10));
print_r(tribonacci([0, 1, 1], 10));

File: Python/6py 3/tribonacci.py
"""
|--------------------------------------------------------------------------
| Tribonacci Sequence
|--------------------------------------------------------------------------
|
"""

def tribonacci(signature, n):
   if n == 0: return []
   res = signature[:n]
   while len(res) < n:
      x = sum(res[-3:])
      res.append(x)
   return res


def tribonacci(signature,n):
   return signature[:n] if n<=len(signature) else tribonacci(signature + [sum(signature[-3:])],n)


def tribonacci(s, n):
   for i in range(3, n): s.append(s[i-1] + s[i-2] + s[i-3])
   return s[:n]


print(tribonacci([1, 1, 1], 10), [1, 1, 1, 3, 5, 9, 17, 31, 57, 105])
print(tribonacci([0, 0, 1], 10), [0, 0, 1, 1, 2, 4, 7, 13, 24, 44])
print(tribonacci([0, 1, 1], 10), [0, 1, 1, 2, 4, 7, 13, 24, 44, 81])